common/src/main/java/org/wso2/testgrid/common/util/TestGridUtil.java

```diff
@@ -158,7 +160,14 @@ public static String executeCommand(String command, File workingDirectory, Deplo
      */
     public static String getTestGridHomePath() {
         String testGridHome = EnvironmentUtil.getSystemVariableValue(TESTGRID_HOME_ENV);
-        Path testGridHomePath = Paths.get(testGridHome);
+        Path testGridHomePath;
+        if (testGridHome == null) {
+            logger.warn("TESTGRID_HOME environment variable not set. Defaulting to ~/.testgrid.");
+            testGridHomePath = DEFAULT_TESTGRID_HOME;
```

Comment:
What if the default directory is not available. IMO we should create a directory if not exists. WDYT?

Reply:
Then, it will get created via https://github.com/wso2/testgrid/blob/master/core/src/main/java/org/wso2/testgrid/core/command/GenerateTestPlanCommand.java#L298

But, it makes sense to create the directory here itself. Added!